jobbergate-core/jobbergate_core/tools/requests.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from pathlib import Path
+from textwrap import dedent
+from typing import Any, Type, TypeVar
+
+from buzz import check_expressions, handle_errors
+from httpx import Client, HTTPStatusError, RequestError, Response, Request
+from loguru import logger
+from pydantic import BaseModel
+
+
+class JobbergateRequestError(RequestError):
+    """
+    Jobbergate specific exceptions that may occur when preparing a request.
+    """
+
+
+class RequestModelError(JobbergateRequestError):
+    """
+    An error occurred while preparing the request data from a model.
+    """
+
+
+class JobbergateResponseError(HTTPStatusError):
+    """
+    Jobbergate specific exceptions that may occur when handling a response.
+    """
+
+
+ResponseModel = TypeVar("ResponseModel", bound=BaseModel)
+
+
+def deserialize_request_model(request_model: BaseModel, request_kwargs: dict[str, Any]):
+    """
+    Deserialize a pydantic model instance into request_kwargs for an httpx client request in place.
+    """
+    with check_expressions(
+        main_message="Request was incorrectly structured to include a request model instance.",
```

Comment:
This message is confusing. It makes it sound like the error was that a model instance was included in the request.

Reply:
Solved in [f706334](https://github.com/omnivector-solutions/jobbergate/pull/587/commits/f70633446ba994f2ada899d3e02ffa2fafc8ab6e)